HISTORY.md

```diff
@@ -2,7 +2,7 @@
 ## 6.23.0 (2021-07-16)
 ### Behavior Changes
 * Obsolete keys in the bottommost level that were preserved for a snapshot will now be cleaned upon snapshot release in all cases. This form of compaction (snapshot release triggered compaction) previously had an artificial limitation that multiple tombstones needed to be present.
-
+* Allow the options.wal_filter to be blank.  If the wal_filter is not specified, the system will store the WALs in the database directory.  This change allows databases to be relocated and opened without updating the corresponding Options files.
```

Comment:
s/wal_filter/wal_dir/ but I don't think there is such a behavior change. wal_dir could previously be empty (default) and WALs stored in DB directory.

This is only a change (bug fix I would argue) to APIs like `LoadLatestOptions` or `ldb --try_load_options`.

Or I guess if the client is reading wal_dir from `GetOptions()`, it might now be empty instead of sanitized to be set to db directory. Again, that is a change to `GetOptions()`, not to `DB::Open()`. For this case we should explicitly recommend change to `.GetWalDir()` from `.wal_dir`.

Reply:
From the client perspective, the only thing that changes is GetOptions().wal_dir will not be sanitized/updated and will (more often than not) be left as the value passed in originally.  Internally, this change allows the DB directory to be relocated under certain circumstances.

The .GetWalDir method is not available as part of the public API.  When reading WAL files (GetCurrent/SortedWalFile, the path is part of the LogFile information returned.  If requested (and if not already present), I can add tests calling these methods with different settings of wal_dir.

Reply:
Where would you like this to appear in the HISTORY?  The change should allow a database to be moved and still opened via LoadLatestOptions (well, if db_log_dir is not set).  Or it can simply be limited to fixing ldb --try_load_options from the HISTORY perspective.

Reply:
We have evidence of users considering this an unexpected, bad behavior, so I would call it a bug fix. Suggestion:

> Fixed handling of `DBOptions::wal_dir` with `LoadLatestOptions()` or `ldb --try_load_options` on a copied or moved DB. Previously, when the WAL directory is same as DB directory (default), a copied or moved DB would reference the old path of the DB as the WAL directory, potentially corrupting both copies. Under this change, the `wal_dir` from `DB::GetOptions()` or `LoadLatestOptions()` can now be empty, which indicates that the *current* DB directory is used for WALs. This is also a subtle API change.